src/python_testing/TC_MWOM_1_2.py

```diff
@@ -0,0 +1,113 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_MWOM_1_2(MatterBaseTest):
+
+    async def read_mod_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.MicrowaveOvenMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_MWOM_1_2(self) -> str:
+        return "[TC-MWOM-1.2] Cluster attributes with DUT as Server"
+
+    def steps_TC_MWOM_1_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read the SupportedModes attribute"),
+            TestStep(3, "Read the CurrentMode attribute"),
+        ]
+        return steps
+
+    def pics_TC_MWOM_1_2(self) -> list[str]:
+        pics = [
+            "MWOM.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_MWOM_1_2(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        attributes = Clusters.MicrowaveOvenMode.Attributes
+
+        self.step(1)
+
+        self.step(2)
+        supported_modes = await self.read_mod_attribute_expect_success(endpoint=endpoint, attribute=attributes.SupportedModes)
+        asserts.assert_greater_equal(len(supported_modes), 2, "SupportedModes must have at least 2 entries!")
+        asserts.assert_less_equal(len(supported_modes), 255, "SupportedModes must have at most 255 entries!")
+        modes = []
```

Comment:
This thing throws me in every test I see it in because the word mode is used so many times.

ugh, and I feel dirty for saying this, but does this work? It's more "pythonic".
```
modes = set([m.mode for m in supported_modes])
assert.assert_equal(len(modes), len(supported_modes), "SupportedModes must have unique mode values")
```
ditto below

Reply:
Seems like it should work.